contracts/oracles/CurveLPOracle.sol

```diff
@@ -29,22 +34,43 @@ contract CurveLPOracle is OracleSimple {
 
     // CurveProvider contract
     CurveProvider public immutable curveProvider;
+    // ChainlinkedOracle contract
+    ChainlinkedOracleSimple public immutable chainlinkedOracle;
 
     /**
      * @param _curveProvider The address of the Curve Provider. Mainnet: 0x0000000022D53366457F9d5E68Ec105046FC4383
+     * @param _chainlinkedOracle The address of the Chainlinked Oracle
      **/
-    constructor(address _curveProvider) {
-        require(_curveProvider != address(0), "Unit Protocol: ZERO_ADDRESS");
+    constructor(address _curveProvider, address _chainlinkedOracle) {
+        require(_curveProvider != address(0) && _chainlinkedOracle != address(0), "Unit Protocol: ZERO_ADDRESS");
         curveProvider = CurveProvider(_curveProvider);
+        chainlinkedOracle = ChainlinkedOracleSimple(_chainlinkedOracle);
     }
 
     // returns Q112-encoded value
     function assetToUsd(address asset, uint amount) public override view returns (uint) {
         if (amount == 0) return 0;
         CurveRegistry cR = CurveRegistry(curveProvider.get_registry());
-        require(cR.get_pool_from_lp_token(asset) != address(0), "Unit Protocol: NOT_A_CURVE_LP");
+        CurvePool cP = CurvePool(cR.get_pool_from_lp_token(asset));
+        require(address(cP) != address(0), "Unit Protocol: NOT_A_CURVE_LP");
+        require(ERC20Like(asset).decimals() == 18, "Unit Protocol: INCORRECT_DECIMALS");
+
+        uint coinsCount = cR.get_n_coins(address(cP));
+        require(coinsCount != 0, "Unit Protocol: CURVE_INCORRECT_COINS_COUNT");
+
+        uint minEthCoinPrice_q112;
+
+        for (uint i = 0; i < coinsCount; i++) {
+            uint ethCoinPrice_q112 = chainlinkedOracle.assetToEth(cP.coins(i), PRECISION);
+            if (minEthCoinPrice_q112 == 0 || ethCoinPrice_q112 < minEthCoinPrice_q112) {
+                minEthCoinPrice_q112 = ethCoinPrice_q112;
+            }
+        }
+
+        uint minUsdCoinPrice_q112 = chainlinkedOracle.ethToUsd(minEthCoinPrice_q112) / PRECISION;
+
+        uint price_q112 = cP.get_virtual_price() * minUsdCoinPrice_q112 / PRECISION;
 
```

Comment:
Also in the above two lines... You don't divide by precision twice by any chance?
E.g. does `minUsdCoinPrice_q112` need `/PRECISION`?

Reply:
Example:
```
minEthCoinPrice_q112 = 1e18 / 1900 (inverse price in ETH)
minUsdCoinPrice_q112 = 1e18 / 1900 / 1e18 = 0
rekt
```

Reply:
// unless I misunderstand what's returned over there